### PR TITLE
Fixes #25628: Resolve repo set id by org/product

### PR DESCRIPTION
**Pull request by akofink**
Resolve repository set IDs from repository set name and one of organization or product. Require organization or product if repository set name is provided.

Add repository set tests.

### CODE REVIEW

File: lib/hammer_cli_katello/search_options_creators.rb [hunk 1]
```diff
@@ -4,6 +4,13 @@ module HammerCLIKatello
   module SearchOptionsCreators
     include HammerCLIKatello::ForemanSearchOptionsCreators
 
+    def create_repository_sets_search_options(options, mode = nil)
+      create_search_options(options, api.resource(:repository_sets), mode).merge(
+        'product_id' => options[HammerCLI.option_accessor_name('product_id')],
+        'organization_id' => options[HammerCLI.option_accessor_name('organization_id')]
+      )
```
Comment (akofink) [inline lib/hammer_cli_katello/search_options_creators.rb]: This is the change that actually fixes the bug. The validate_options above is for giving nice error messages to the user if they use the wrong combination of options.

File: lib/hammer_cli_katello/repository_set.rb [hunk 2]
```diff
@@ -29,6 +29,15 @@ class InfoCommand < HammerCLIKatello::InfoCommand
...
       end
 
+      validate_options do
+        if option(:option_name).exist?
+          any(
+            :option_organization_id, :option_organization_name, :option_organization_label,
+            :option_product_id, :option_product_name
+          ).required
+        end
+      end
```
[the remark this answers is not in the document]
Comment (akofink) [reply lib/hammer_cli_katello/repository_set.rb]: I was playing around with this idea, trying to put it like this:

```rb
module HammerCLIKatello
  class RepositorySetCommand < HammerCLI::AbstractCommand
    def self.validate_repo_set
      validate_options do
        ...
      end
    end

    class ListCommand < HammerCLIKatello::ListCommand
      RepositorySetCommand.validate_repo_set
    end
  end
end
```

But the `options(:product_name)` and the like were not working properly. I will try putting it in a module and including it though.

[the remark this answers is not in the document]
Comment (akofink) [reply lib/hammer_cli_katello/repository_set.rb]: I got it to work by [extending a module](https://github.com/Katello/hammer-cli-katello/pull/606/files#diff-ef0fadba24fe741385d16e27a1aaac45R3).

**Comment by akofink** [comment]
Updated with changes from https://github.com/Katello/hammer-cli-katello/pull/608

**Comment by beav** [comment]
@akofink looks like this needs a squash to make the prprocessor test happy.

The code looks ok, testing now

**Comment by akofink** [comment]
Thanks @beav! That pesky 3.11 test data commit is making prprocessor unhappy; I'll make sure to squash 'n merge. @cfouant did you want to review as well?

**Pull request merged**